velox/type/Filter.cpp

```diff
@@ -779,4 +779,240 @@ std::unique_ptr<Filter> BigintMultiRange::mergeWith(const Filter* other) const {
       VELOX_UNREACHABLE();
   }
 }
+
+std::unique_ptr<Filter> BytesRange::mergeWith(const Filter* other) const {
+  switch (other->kind()) {
+    case FilterKind::kAlwaysTrue:
+    case FilterKind::kAlwaysFalse:
+    case FilterKind::kIsNull:
+      return other->mergeWith(this);
+    case FilterKind::kIsNotNull:
+      return std::make_unique<BytesRange>(*this, false);
```

Comment:
Nit: I will suggest to add /*nullAllowed=*/ for `false` argument (which will look like `return std::make_unique<BytesRange>(*this, /*nullAllowed=*/false);`

Basically this will increase the readability for a constant assignment of an argument. Same for other cases in this PR.